File: CSharp/Tests/SingleNumber.cs
using Solutions;

namespace Tests;

public class SingleNumberTest {
    [Fact]
    public void SingleNumber_Case1() {
        int[] nums = [2, 2, 1];
        int want = 1;

        int result = SingleNumber.SingleNumber1(nums);
        Assert.Equal(want, result);
    }

    [Fact]
    public void SingleNumber_Case2() {
        int[] nums = [4, 1, 2, 1, 2];
        int want = 4;

        int result = SingleNumber.SingleNumber1(nums);
        Assert.Equal(want, result);
    }

    [Fact]
    public void SingleNumber_Case3() {
        int[] nums = [1];
        int want = 1;

        int result = SingleNumber.SingleNumber1(nums);
        Assert.Equal(want, result);
    }
}

File: Rust/tests/single_number.rs
#[cfg(test)]
mod tests {
    use leetcode::single_number::*;

    #[test]
    fn test_single_number_case1() {
        let nums = vec![2, 2, 1];
        let want = 1;

        let result = single_number(nums);

        assert_eq!(result, want);
    }

    #[test]
    fn test_single_number_case2() {
        let nums = vec![4, 1, 2, 1, 2];
        let want = 4;

        let result = single_number(nums);

        assert_eq!(result, want);
    }

    #[test]
    fn test_single_number_case3() {
        let nums = vec![1];
        let want = 1;

        let result = single_number(nums);

        assert_eq!(result, want);
    }
}
